@here/harp-mapview/lib/Tile.ts

```diff
@@ -492,56 +487,43 @@ export class Tile implements CachedResource {
     }
 
     /**
+     * @internal
+     * @deprecated
+     *
      * Gets the list of developer-defined [[TextElement]] in this `Tile`. This list is always
      * rendered first.
      */
     get userTextElements(): TextElementGroup {
-        return this.m_userTextElements;
+        return (
+            this.m_textElementGroups.groups.get(Number.MAX_SAFE_INTEGER) ??
+            new TextElementGroup(Number.MAX_SAFE_INTEGER)
+        );
     }
 
     /**
      * Adds a developer-defined [[TextElement]] to this `Tile`. The [[TextElement]] is always
      * visible, if it's in the map's currently visible area.
      *
+     * @deprecated use [[addTextElement]].
+     *
      * @param textElement The Text element to add.
      */
     addUserTextElement(textElement: TextElement) {
-        if (this.m_textElementsChanged === false) {
-            // HARP-8733: Text content in the tile is about to change, but it has already been
-            // rendered at least once (otherwise m_textElementsChanged would be undefined). Clone
-            // the text element group so that it's handled as a new group by TextElementsRenderer
-            // and it doesn't reuse the same state stored for the old one.
-            // TODO: HARP-8910 Deprecate user text elements.
-            this.m_userTextElements = this.m_userTextElements.clone();
-        }
-
-        this.m_userTextElements.elements.push(textElement);
-        this.textElementsChanged = true;
+        textElement.priority = Number.MAX_SAFE_INTEGER;
```

Comment:
please move Number.MAX_SAFE_INTEGER to a named constant, so that it has some meaning, especially above in the userTextElements function.

Reply:
done